pkg/base1/cockpit.js

```diff
@@ -926,6 +926,29 @@ function basic_scope(cockpit) {
         });
     };
 
+    /* Not public API ... yet? */
+    cockpit.expect_restart = function expect_restart(delay, host) {
+        if (!host)
+            host = default_host;
+
+        var timer = null;
+        var options = { "command": "expect-restart",
+                        "host": host };
+
+        function send() {
+            ensure_transport(function(transport) {
+                transport.send_control(options);
+                if (timer)
+                    window.clearTimeout(timer);
```

Comment:
What is happening here? Why clearTimeout? Hasn't the timer already fired at this point?